pandas/core/categorical.py

```diff
@@ -443,12 +443,18 @@ def _validate_categories(cls, categories):
             raise ValueError('Categorical categories must be unique')
         return categories
 
-    def _set_categories(self, categories):
+    def _set_categories(self, categories, validate=True):
         """ Sets new categories """
```

Comment:
My example came from the test suite :)

There's about 8 tests that fail / error if I change the constructor to skip category validation when `fastpath` is true. Not sure if I'll be able to get to those today.